zrml/global-disputes/src/tests.rs

```diff
@@ -106,10 +111,59 @@ fn add_vote_outcome_works() {
     });
 }
 
+#[test]
+fn add_vote_outcome_fails_with_outcome_mismatch() {
+    ExtBuilder::default().build().execute_with(|| {
+        // create scalar market
+        let market_id = 0u128;
+        let market = market_mock::<Runtime>();
+        Markets::<Runtime>::insert(market_id, &market);
+        GlobalDisputes::push_voting_outcome(
+            &market_id,
+            OutcomeReport::Scalar(0),
+            &ALICE,
+            10 * BASE,
+        )
+        .unwrap();
+        assert_noop!(
+            GlobalDisputes::add_vote_outcome(
+                Origin::signed(ALICE),
+                market_id,
+                OutcomeReport::Categorical(0u16),
+            ),
+            Error::<Runtime>::OutcomeMismatch
+        );
+    });
+}
+
+#[test]
+fn add_vote_outcome_fails_with_non_existing_market() {
+    ExtBuilder::default().build().execute_with(|| {
+        let market_id = 0u128;
+        GlobalDisputes::push_voting_outcome(
+            &market_id,
+            OutcomeReport::Scalar(0),
+            &ALICE,
+            10 * BASE,
+        )
+        .unwrap();
```

Comment:
This should fail. I mean... why would it be okay to push an outcome for a non-existent market?

Reply:
True, it is just like, that we are in full control of the private API calls like `push_voting_outcome` and `determine_voting_winner`. So it's not really needed, because the `market` check is always inside `prediction-markets` before the private API calls.

Reply:
Depends on what you mean by private, I guess. If you think of `zrml_global_disputes` as the object, then the object exposes an API to other pallets. These pallets are now able to push different types of outcome tokens to a global dispute, which is a violation of the rules of `zrml_global_disputes`. Fans of OOP will tell you that's dangerous. In general, I would agree with that sentiment.

That being said, OOP isn't always perfect. There are reasons why an API would not check the input. Sometimes its too costly, sometimes virtually impossible. It's your call.

Reply:
b14faafad72aa89502de65d85024b4c8e86d95d6